src/explore-education-statistics-common/src/components/form/FormFieldCheckboxGroup.tsx

```diff
@@ -1,53 +1,81 @@
-import FormField, {
-  FormFieldComponentProps,
-} from '@common/components/form/FormField';
-import { OmitStrict } from '@common/types';
+import FormCheckboxGroup, {
+  CheckboxOption,
+  FormCheckboxGroupProps,
+} from '@common/components/form/FormCheckboxGroup';
+import { useFormIdContext } from '@common/components/form/contexts/FormIdContext';
+import useRegister from '@common/components/form/hooks/useRegister';
+import handleAllCheckboxChange from '@common/components/form/util/handleAllCheckboxChange';
+import getErrorMessage from '@common/components/form/util/getErrorMessage';
 import React from 'react';
-import FormCheckboxGroup, { FormCheckboxGroupProps } from './FormCheckboxGroup';
-import handleAllChange from './util/handleAllCheckboxChange';
+import { FieldValues, Path, useFormContext, useWatch } from 'react-hook-form';
 
-type Props<FormValues> = OmitStrict<
-  FormFieldComponentProps<FormCheckboxGroupProps, FormValues>,
-  'formGroup'
->;
+export interface Props<TFormValues extends FieldValues>
+  extends Omit<FormCheckboxGroupProps, 'name' | 'value' | 'id'> {
+  name: Path<TFormValues>;
+  id?: string;
+  options: CheckboxOption[];
+  showError?: boolean;
+  disabled?: boolean;
+}
+
+export default function FormFieldCheckboxGroup<
+  TFormValues extends FieldValues,
+>({
+  name,
+  id: customId,
+  options,
+  showError = true,
+  disabled = false,
+  ...props
+}: Props<TFormValues>) {
+  const {
+    formState: { errors },
+    register,
+    setValue,
+    trigger,
+  } = useFormContext<TFormValues>();
 
-function FormFieldCheckboxGroup<FormValues>(props: Props<FormValues>) {
-  const { options, onChange, onAllChange } = props;
+  const { ref: inputRef, ...field } = useRegister(name, register);
+  const { fieldId } = useFormIdContext();
+  const id = fieldId(name, customId);
+  const selectedValues = useWatch({ name }) || [];
+  const { onAllChange, onChange } = props;
```

Comment:
Most minuscule of trivialities, but is there any reason we're de-structuring these props here instead of up with the others?

Reply:
I think it's because onChange and onAllChange still want to be passed down to the FormCheckbox component when we spread the props in, but, as you say, it's pre-exisitng so I'm not sure.